buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/rules/AbstractGradleViolationRule.groovy

```diff
@@ -128,7 +132,7 @@ abstract class AbstractGradleViolationRule extends AbstractContextAwareViolation
                 return sourceFile
             }
         }
-        throw new RuntimeException("No source file found for: $className")
+        return null
```

Comment:
Why this change as part of this PR?

Reply:
If we don't have this change, then we get an NPE when running the binary compatibility check: The class `CompatibilityAdapterForTaskInputs` on the public API has been removed, so we try to get the source code for it and file with an exception if we don't find it. I guess the check could be more clever, like only trying to get the source file if the change is not a removal. WDYT?

Reply:
Sounds like doing so would make the use case more explicit in the code, too. I'd say let's go for it.

Reply:
Done.